connection.go

```diff
@@ -1031,7 +1040,10 @@ func (c *Connection) input(p []byte) error {
 		}
 	}
 
-	return err
+	if remainder != nil {
+		return c.input(remainder)
```

Comment:
Note that this would not work properly if you actually were checking the length of CI.